### PR TITLE
PROD-750 sam client timeout and error handling

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: <Link to Jira ticket>
<Put notes here to help reviewer understand this PR>

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/WorkspaceManagerResourceMonitorRecordDao.scala [hunk 1]
```diff
@@ -11,16 +11,17 @@ import scala.concurrent.Future
...
+  def create(jobRecordId: UUID, jobType: JobType, billingProjectName: String, userEmail: RawlsUserEmail): Future[Unit] =
+    create(
+      WorkspaceManagerResourceMonitorRecord(
+        jobRecordId,
+        jobType,
+        None,
+        Some(billingProjectName),
+        Some(userEmail.value),
+        Timestamp.from(Instant.now())
+      )
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/WorkspaceManagerResourceMonitorRecordDao.scala:23]: scalafmt

File: core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BootMonitors.scala [hunk 2]
```diff
@@ -191,15 +190,14 @@ object BootMonitors extends LazyLogging {
...
-        workspaceManagerDAO,
-        gcsDAO
-      )
+      system,
+      conf,
+      slickDataSource,
+      samDAO,
+      workspaceManagerDAO,
+      gcsDAO
+    )
```
Comment (dvoet) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BootMonitors.scala:200]: scalafmt

File: core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingRepositorySpec.scala [hunk 3]
```diff
@@ -3,7 +3,16 @@ package org.broadinstitute.dsde.rawls.billing
...
+import org.broadinstitute.dsde.rawls.model.{
+  CreationStatuses,
+  CromwellBackend,
+  GoogleProjectNumber,
+  RawlsBillingAccountName,
+  RawlsBillingProject,
+  RawlsBillingProjectName,
+  RawlsUserEmail,
+  ServicePerimeterName
+}
```
Comment (dvoet) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BillingRepositorySpec.scala:15]: scalafmt

File: core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitorSpec.scala [hunk 4]
```diff
@@ -4,7 +4,10 @@ import org.broadinstitute.dsde.rawls.TestExecutionContext
 import org.broadinstitute.dsde.rawls.dataaccess.WorkspaceManagerResourceMonitorRecordDao
 import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.{JobStatus, JobType}
 import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.JobType.JobType
-import org.broadinstitute.dsde.rawls.dataaccess.slick.{WorkspaceManagerResourceJobRunner, WorkspaceManagerResourceMonitorRecord}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{
+  WorkspaceManagerResourceJobRunner,
+  WorkspaceManagerResourceMonitorRecord
+}
```
Comment (dvoet) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/WorkspaceResourceMonitorSpec.scala:10]: scalafmt

**Review by rtitle — APPROVED** [review]
Wow nice find.. yeah it makes sense that an uncompleted promise could lead to things hanging. Looks like this adds 2 protections: error handling and timeouts. The code looks good to me 👍 

(Maybe there are some test compilation errors though, just looking at github action results)

**Review by MatthewBemis — APPROVED** [review]

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest

**Comment by dvoet** [comment]
jenkins retest

**Comment by jgainerdewar** [comment]
jenkins retest

**Pull request merged**